File: sample/src/main/java/tech/thdev/androidrecyclerview/view/animation/AnimatorEvent.java
package tech.thdev.androidrecyclerview.view.animation;

import android.animation.Animator;

import io.reactivex.Observable;

/**
 * Created by Tae-hwan on 20/10/2016.
 */

public class AnimatorEvent {

    public static Observable<ItemAnimatorEvent> eventObservable(Animator animator) {
        return Observable.create(new AnimatorEventSubScribe(animator));
    }
}

File: sample/src/main/java/tech/thdev/androidrecyclerview/ui/animation/AnimatorEvent.kt
package tech.thdev.androidrecyclerview.ui.animation

import android.animation.Animator
import io.reactivex.rxjava3.core.BackpressureStrategy
import io.reactivex.rxjava3.core.Flowable

/**
 * Created by Tae-hwan on 20/10/2016.
 */
object AnimatorEvent {
    fun eventObservable(animator: Animator): Flowable<ItemAnimatorEvent> {
        return Flowable.create(AnimatorEventSubScribe(animator), BackpressureStrategy.BUFFER)
    }
}